booking.yaml

```diff
@@ -0,0 +1,516 @@
+      required:
+        - id
+        - firstName
+        - lastName
+      properties:
+        id:
+          description: The identifier MaaS uses to identify the customer
+          type: string
+        firstName:
+          description: First name of the customer (e.g. John)
+          type: string
+        lastName:
+          description: Last name of the customer (e.g. Doe)
+          type: string
+        phone:
+          description: Phone number that the customer may be reached from
+          type: string
+    options:
+      type: object
+      description: Containing an array of available options matching the query
+      properties:
+        leg:
+          $ref: '#/components/schemas/options_leg'
+        meta:
+          $ref: '#/components/schemas/options_meta'
+    options_leg:
+      type: object
+      properties:
+        startTime:
+          $ref: '#/components/schemas/time'
+        endTime:
+          $ref: '#/components/schemas/time'
+        from:
+          $ref: '#/components/schemas/options_coordinates'
+        to:
+          $ref: '#/components/schemas/options_coordinates'
+    options_coordinates:
+      type: object
+      properties:
+        lat:
+          $ref: '#/components/schemas/lat'
+        lon:
+          $ref: '#/components/schemas/lon'
+    options_meta:
+      type: object
+      properties:
+        name:
+          type: string
+        description:
+          type: string
+        image:
+          type: string
+          format: url
+        car:
+          type: object
+          properties:
+            passengers:
+              type: integer
+    leg:
+      type: object
+      description: >-
+        A OpenTripPlanner compatible definition of a leg (see OpenTripPlanner
+        docs for reference)
+      additionalProperties: true
+      properties:
+        from:
+          description: The coordinates the TSP should use to resolve leg start location
+          $ref: '#/components/schemas/place'
+        to:
+          description: The coordinates the TSP should use to resolve leg finish location
+          $ref: '#/components/schemas/place'
+        startTime:
+          $ref: '#/components/schemas/time'
+        endTime:
+          $ref: '#/components/schemas/time'
+        mode:
+          $ref: '#/components/schemas/mode'
+        departureDelay:
+          $ref: '#/components/schemas/duration'
+        arrivalDelay:
+          $ref: '#/components/schemas/duration'
+        distance:
+          $ref: '#/components/schemas/distance'
+        fare:
+          $ref: '#/components/schemas/fare'
+        route:
+          type: string
+        routeShortName:
+          type: string
+        routeLongName:
+          type: string
+        agencyId:
+          type: string
+        legGeometry:
+          $ref: '#/components/schemas/legGeometry'
+      required:
+        - from
+        - to
+        - mode
+        - startTime
+        - endTime
+    place:
+      type: object
```

Comment:
Can reuse [Location](https://github.com/maas-alliance/apis/blob/master/maas-api.yaml#L2436) and [StopLocation](https://github.com/maas-alliance/apis/blob/master/maas-api.yaml#L1202)

Reply:
Cool. How should we separate those definitions, so they can be shared between APIs?

Reply:
Enhancement opened:
https://github.com/maas-alliance/apis/issues/11